packages/ava/src/text/tools/getVariables.ts

```diff
@@ -0,0 +1,57 @@
+import type { BulletsParagraphSpec, EntityPhraseSpec, NarrativeTextSpec, ParagraphSpec, PhraseSpec } from '../schema';
+
+// extract the variables from the schema
+export const getVariableMap = (spec: NarrativeTextSpec) => {
```

Comment:
ok, I think normal function is better.

@chenluli `getVariableMap` is not used, can we depraced it?

Reply:
OK, this function was used in a demo page to collect all variables, but it can be deprecated now because we might need to redesign the method for sourcing all variables. Do we need to deprecate "sourceId" in metadata this time?

Reply:
> OK, this function was used in a demo page to collect all variables, but it can be deprecated now because we might need to redesign the method for sourcing all variables. Do we need to deprecate "sourceId" in metadata this time?

ok, I handle sourceId by this way 👉 https://github.com/antvis/AVA/pull/477/files#diff-8f7b752ed6fda1399c045d3a6f7e8dc3179671d714ab65faffe712e88fcbe028R102-R104

Reply:
👌